### PR TITLE
Pull DetailView into App (Closes #24)

### LINKED ISSUE #24: DetailView outside MapChart
I need the detailview to be outside of the mapchart — to make the animations and size smoother.

Currently, the HTML structure looks something like this:

```
    <App>
        <Title/>
        <Subtitle/>
        <MapChart>
            <DetailView/>
        </MapChart>
    </App>
```

I want DetailView outside of the MapChart, so it looks more like this:

```
    <App>
        <Title/>
        <Subtitle/>
        <MapChart/>
        <DetailView/>
    </App>
```

### PR DESCRIPTION
**Pull request by dvargas92495**
PR for knowledge sharing ;)

I would suggest reviewing with hiding white space changes:
image

### CODE REVIEW

File: components/App.js [hunk 1]
```diff
@@ -149,7 +150,8 @@ class App extends React.Component {
               </Typist>
             </div>
 
-            <MapChart setMarker={this.setMarker} />
+            <MapChart marker={this.state.marker} setMarker={this.setMarker} />
+            <DetailView marker={this.state.marker} setMarker={this.setMarker} />
```
Comment (dvargas92495) [inline components/App.js:154]: `App` is now the only component managing marker state

File: components/DetailView.js [hunk 2]
```diff
@@ -195,7 +195,7 @@ class DetailView extends React.Component {
 
 DetailView.propTypes = {
   marker: PropTypes.object,
-  onClose: PropTypes.func.isRequired
+  setMarker: PropTypes.func.isRequired
```
Comment (dvargas92495) [inline components/DetailView.js:198]: this makes me sad that we're not using TypeScript

File: components/MapChart.js [hunk 3]
```diff
@@ -16,22 +15,15 @@ const geoUrl = "https://cdn.jsdelivr.net/npm/us-atlas@3/states-10m.json";
...
-
   render() {
-    const marker = markers[this.state.marker];
     return (
       <>
         <ComposableMap width={1200} projection="geoAlbersUsa"
           style={{
             transition: "width 0.5s ease-in-out",
-            width: marker ? '300px' : '1000px'
+            width: this.props.marker ? '300px' : '1000px'
```
Comment (dvargas92495) [inline components/MapChart.js:26]: this might get pulled out with the CSS changes, left for now for limiting scope

**Comment by dvargas92495** [comment]
It looks like we're auto deploying branches to https://chamilea-hkzcruxr9.vercel.app/. Neat

**Comment by kennethshawfriedman** [comment]
Thank you! This mostly makes sense to me... but I should probably just make a small React app from scratch to get a feel for how the data passing works.

File: components/App.js (hunk 1 above)
Comment (kennethshawfriedman) [reply components/App.js:154]: 👍

**Review by kennethshawfriedman — APPROVED** [review]

**Pull request merged**